measure.rb

```diff
@@ -521,12 +521,37 @@ def self.explode_surfaces(runner, model)
     end
     max_azimuth_length = azimuth_lengths.values.max
 
+    success = add_neighbors(runner, model, building, max_azimuth_length)
+    return false if not success
+
     # Initial distance of shifts at 90-degrees to horizontal outward
     azimuth_side_shifts = {}
     azimuth_lengths.each do |key, value|
       azimuth_side_shifts[key] = max_azimuth_length / 2.0
     end
 
+    # Explode neighbors
+    model.getShadingSurfaceGroups.each do |shading_surface_group|
+      next if shading_surface_group.name.to_s != Constants.ObjectNameNeighbors
+
+      shading_surface_group.shadingSurfaces.each do |shading_surface|
+        azimuth = shading_surface.additionalProperties.getFeatureAsInteger("Azimuth").get
+        azimuth_rad = UnitConversions.convert(azimuth, "deg", "rad")
+        distance = shading_surface.additionalProperties.getFeatureAsDouble("Distance").get
+
+        unless azimuth_lengths.keys.include? azimuth
+          runner.registerError("A neighbor building surface has an azimuth '#{azimuth}' not equal to the amizuth of any wall (#{azimuth_lengths.keys * ", "}).")
```

Comment:
Let's say "neighbor building has an azimuth" instead of "neighbor building surface has an azimuth".